neqo-transport/src/cc/classic_cc.rs

```diff
@@ -73,98 +72,59 @@ impl State {
 }
 
 #[derive(Debug)]
-pub struct CongestionControl {
+pub struct ClassicCongestionControl {
+    cc_algorithm: Box<dyn CwndFn>,
```

Comment:
I think that you can template this one.  That is `ClassCongestionControl<T>` and condition the implementation on `T:CwndFn`.